tubes/lua.lua

```diff
@@ -464,11 +464,25 @@ local safe_globals = {
 	"tonumber", "tostring", "type", "unpack", "_VERSION"
 }
 
-local function create_environment(pos, mem, event, itbl, send_warning)
+local function create_environment(pos, mem, event, stack, itbl, send_warning)
 	-- Make sure the tube hasn't broken.
 	local vports = minetest.registered_nodes[minetest.get_node(pos).name].virtual_portstates
 	if not vports then return {} end
 
+	local function get_item_tag()
+		if not pipeworks.enable_item_tags then return nil end
+		return pipeworks.get_item_tag(stack)
+	end
+	local function set_item_tag(tag)
+		if not pipeworks.enable_item_tags and not stack then return end
```

Comment:
No need to check the setting again.
```suggestion
		if not stack then return end
```